src/clib/gaussj_g.cpp

```diff
@@ -0,0 +1,76 @@
+#include <numeric>
+#include <vector>
+
+#include "gaussj_g.hpp"
+
+namespace grackle::impl {
+
+int gaussj_g(int n, double* coef_matrix_fortran, double* vector) {
+
+    // TODO: to be removed
+    // Copy the matrix to a C-style layout (column-major order)
+    std::vector<double> coef_matrix(n*n);
+    for (int i = 0; i < n; ++i) {
+        for (int j = 0; j < n; ++j) {
+        coef_matrix[j*n + i] = coef_matrix_fortran[i*n + j];
+        }
+    }
+
+    // Loop over columns
+    for(unsigned int col=0; col<n; col++) {
+
+        // Find the largest (absolute value) element in the column
+        double max_val = std::abs(coef_matrix[col*n + col]);
+        unsigned int max_row = col;
+        for(unsigned int row = col+1; row < n; row++) {
+            if(std::abs(coef_matrix[row*n + col]) > max_val) {
```

Comment:
```suggestion
            if(std::fabs(coef_matrix[row*n + col]) > max_val) {
```
Not essential, but I think it is less confusing to people more familiar with C than C++